src/main/java/stockservice/StockServiceFactory.java

```diff
@@ -0,0 +1,27 @@
+package stockservice;
+
+import java.util.MissingResourceException;
+
+/**
+ * A factory to that returns the correct concrete class implementation of the 
+ * StockService interface
+ * 
+ * @author Christopher Stafford
+ */
+class StockServiceFactory {
+    /**
+     * 
+     * @param serviceName as <CODE>String</CODE>
+     * @return Correct concrete class for interface StockService 
+     */
+    public static StockService getStockService(String serviceName) {
+        if (serviceName == null) {
```

Comment:
This method should not take an argument. The client should not get to know the implementation of the interface. 
(I realize that what you did here is a common variation of the factory pattern.)

Reply:
Thank you for the feedback. I've removed the arguments and it now just returns the service.